sdk/include/psdk/textserv.h

```diff
@@ -23,9 +23,14 @@
 extern "C" {
 #endif
 
-DEFINE_GUID(IID_ITextServices,0x8d33f740,0xcf58,0x11ce,0xa8,0x9d,0x00,0xaa,0x00,0x6c,0xad,0xc5);
-DEFINE_GUID(IID_ITextHost,    0xc5bdd8d0,0xd26e,0x11ce,0xa8,0x9e,0x00,0xaa,0x00,0x6c,0xad,0xc5);
-DEFINE_GUID(IID_ITextHost2,   0xc5bdd8d0,0xd26e,0x11ce,0xa8,0x9e,0x00,0xaa,0x00,0x6c,0xad,0xc5);
+#ifdef __cplusplus
+#define THISCALLMETHOD_(type,method)  virtual type __thiscall method
+#else
+#define THISCALLMETHOD_(type,method)  type (__thiscall *method)
```

Comment:
I'd assume that `__thiscall` is only valid in c++, not c.

Reply:
I just missed it in this header file, I can define `__thiscall` to be `__stdcall` when compiling with MSVC.  Making that change actually gets it compiling in amd64, but not on i386 MSVC.  Any thoughts on the remaining error?

Reply:
thiscall is C++ only, obviously.

Reply:
Wine is using GCC's builtin support for `__thiscall` (just meaning "first parameter in ecx, rest on the stack") here to implement thiscall functions in C without needing assembly wrappers. It's not C++ only.

Reply:
Ya, this would all be a lot simpler if MSVC supported __thiscall in C code, but that doesn't appear to be the case, so it needs to asm wrappers.